mlflow/artifacts/__init__.py

```diff
@@ -104,7 +106,34 @@ def list_artifacts(
     run_id: Optional[str] = None,
     artifact_path: Optional[str] = None,
     tracking_uri: Optional[str] = None,
-):
+) -> list[FileInfo]:
+    run_artifacts = _list_run_artifacts(
+        artifact_uri=artifact_uri,
+        run_id=run_id,
+        artifact_path=artifact_path,
+        tracking_uri=tracking_uri,
+    )
+    if (
+        run_artifacts
```

Comment:
@harupy What if there's a mix of run artifacts and model artifacts? we should return both...

Reply:
So should we return run_artifacts + model_artifacts?

Reply:
I would prefer listing both runs artifacts and models artifacts. 
I remember we show model artifacts in run's artifacts tab with an instruction:
<img width="1061" alt="image" src="https://github.com/user-attachments/assets/0faf83bd-f110-45d3-bc11-9ee7f2c33fda" />
Does this only work when there are no run artifacts logged? If so I feel like we should fix the UI :)

Reply:
We *must* always show run artifacts *and* model artifacts, or this is a major regression in the UI. The same is true for the API - always return run artifacts *and* model artifacts

Reply:
> I remember we show model artifacts in run's artifacts tab with an instruction:

Model artifacts show up when you only have model artifacts.

Reply:
Yea I just tested https://e2-dogfood.staging.cloud.databricks.com/editor/notebooks/1752476514600810?o=6051921418418893#command/7096535154283234
Seems like we need to fix this UI?